### ISSUE TITLE
clusterctl workflow errors out but everything works

### ISSUE THREAD

**Issue by chuckha** [kubernetes-sigs/cluster-api-provider-aws#313]
/kind bug

**What steps did you take and what happened:**
I followed the getting started guide and ran clusterctl. I received this message:

```
F1025 12:16:59.708945    1402 create_cluster.go:64] unable to update target cluster endpoint: unable to get master IP: ProviderStatus is nil, cannot get IP
```

but everything had already pivoted and was set up perfectly

**What did you expect to happen:**
I expected clusterctl to run without error

**Anything else you would like to add:**
Since no functionality seems to be lost, this is not particularly urgent

**Environment:**
v1.0.0-alpha.2

**Comment by randomvariable** [comment]
This was fixed in pre-CRD migration in #207

**Comment by chuckha** [comment]
for sure, not sure if this was a change in clusterctl or what, but there was an edge case the code didn't handle. incoming PR
